docs/website/blog/2023-11-01-dlt-dagster.md

```diff
@@ -0,0 +1,421 @@
+        issues = response.json()
+        yield issues
+
+        if 'link' in response.headers:
+            if 'rel="next"' not in response.headers['link']:
+                break
+
+            url = response.links['next']['url']  # fetch next page of stargazers
+        else:
+            break
+        time.sleep(2)  # sleep for 2 seconds to respect rate limits
+
+if __name__ == "__main__":
+    # configure the pipeline with your destination details
+    pipeline = dlt.pipeline(
+        pipeline_name='github_issues', destination='bigquery', dataset_name='github_issues_data'
+    )
+
+    # run the pipeline with your parameters
+    load_info = pipeline.run(github_issues_resource())
+
+    #print the information on data that was loaded
+    print(load_info)
+```
+
+The above code creates a simple github_issues pipeline that gets the issues data from the defined repository and loads it into BigQuery. The `dlt.resources` yields the data while the `dlt.pipeline` normalizes the nested data and loads it into the defined destination. To read more about the technical details refer to the `dlt` [docs](https://dlthub.com/docs/intro).
+
+To run the pipeline execute the below commands:
+
+```bash
+pip install -r requirements.txt
+python github_issues.py
+```
+
+We now have a running pipeline and are ready to orchestrate it using dagster.
+
+## **Orchestrating using dagster:**
+
+We will need to adjust our pipeline a bit to orchestrate it using dagster. 
+
+### Step 1: Create a dagster project
+
+- Create a new directory for your Dagster project and scaffold the basic structure:
+
+```bash
+mkdir dagster_github_issues
+cd dagster_github_issues
+dagster project scaffold --name github-issues
+```
+
+This will generate the default files for dagster that we will use as a starting point for our data pipeline. 
+
+### Step 2: Set up the directory Structure
+
+- Inside the `github-issues/github_issues` directory create the following folders: `assets`, `resources`, and `dlt`.
+
+```bash
+.
+├── README.md
+├── github_issues
+│   ├── __init__.py
+│   ├── assets
+│   │   ├── __init__.py
+│   ├── dlt
+│   │   ├── __init__.py
+│   └── resources
+│       ├── __init__.py
+├── github_issues_tests
+│   ├── __init__.py
+│   └── test_assets.py
+├── pyproject.toml
+├── setup.cfg
+└── setup.py
+```
+
+### Step 3: Add dlt Resources and environment variables
+
+- Copy the previously created **`github_issues_resource`** code into **`dlt/__init__.py`** under the **`dlt`** folder. Remove the **`dlt.secrets.value`** parameter, as we'll pass the credentials through a **`.env`** file.
+- Create a `.env` file in the root directory. This is the directory where the `pyproject.toml` file exits. Copy the credentials into the `.env` and follow the correct naming convention. For more info on setting up the `.env` file have a look at the [docs](https://dlthub.com/docs/general-usage/credentials#reading-credentials-from-environment-variables).
```

Comment:
(https://dlthub.com/docs/general-usage/credentials#reading-credentials-from-environment-variables). this link is not valid anymore, it leads to credentials section only